cdap-security/src/main/java/co/cask/cdap/security/guice/SecureStoreServerModule.java

```diff
@@ -0,0 +1,111 @@
+import co.cask.cdap.common.conf.SConfiguration;
+import co.cask.cdap.security.store.DefaultSecureStoreService;
+import co.cask.cdap.security.store.DummySecureStoreService;
+import co.cask.cdap.security.store.FileSecureStoreService;
+import co.cask.cdap.security.store.SecureStoreService;
+import co.cask.cdap.security.store.SecureStoreUtils;
+import co.cask.cdap.security.store.secretmanager.SecretManagerSecureStoreService;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.PrivateModule;
+import com.google.inject.Provider;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import com.google.inject.name.Names;
+
+/**
+ * Server side guice bindings for secure store service related classes.
+ */
+public class SecureStoreServerModule extends PrivateModule {
+  public static final String DELEGATE_SECURE_STORE_SERVICE = "delegateSecureStoreService";
+
+  @Override
+  protected void configure() {
+    bind(SecureStoreService.class)
+      .annotatedWith(Names.named(DELEGATE_SECURE_STORE_SERVICE))
+      .toProvider(SecureStoreServiceProvider.class);
+
+    bind(DefaultSecureStoreService.class).in(Scopes.SINGLETON);
+    bind(SecureStore.class).to(DefaultSecureStoreService.class);
+    bind(SecureStoreManager.class).to(DefaultSecureStoreService.class);
+    bind(SecureStoreService.class).to(DefaultSecureStoreService.class);
+    expose(SecureStore.class);
+    expose(SecureStoreManager.class);
+    expose(SecureStoreService.class);
+  }
+
+  /**
+   * Secure store service provider.
+   *
+   * file binds to FileSecureStoreService
+   * kms  binds to KMSSecureStoreService
+   * ext  binds to SecretManagerSecureStoreService
+   * none binds to DummySecureStoreService
+   *
+   * TODO: simplify these bindings after CDAP-14668 is fixed.
+   */
+  @Singleton
+  private static final class SecureStoreServiceProvider implements Provider<SecureStoreService> {
+    private final CConfiguration cConf;
+    private final SConfiguration sConf;
+    private final Injector injector;
+
+    @Inject
+    private SecureStoreServiceProvider(final CConfiguration cConf, SConfiguration sConf, Injector injector) {
+      this.cConf = cConf;
+      this.sConf = sConf;
+      this.injector = injector;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public SecureStoreService get() {
+      if (SecureStoreUtils.isFileBacked(cConf)) {
+        if (Strings.isNullOrEmpty(sConf.get(Constants.Security.Store.FILE_PASSWORD))) {
+          throw new IllegalArgumentException("File secure store password is not set. Please set the " +
+                                               "\"security.store.file.password\" property in cdap-security.xml.");
+        }
+        return injector.getInstance(FileSecureStoreService.class);
+      }
+
+      if (SecureStoreUtils.isKMSBacked(cConf)) {
+        if (SecureStoreUtils.isKMSCapable()) {
```

Comment:
Do it in a consistent style. The `if` for the file backed case is a reverse than in here.

Reply:
This is because wanted to avoid negation. Changed ordering.